pkg/router/controller/controller.go

```diff
@@ -97,7 +114,21 @@ func (c *RouterController) HandleEndpoints() {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
+	// Change the local sync state within the lock to ensure that all
+	// event handlers have the same view of sync state.
+	c.endpointsReplaceRead = c.EndpointsReplaceRead()
```

Comment:
I think resync can be useful when there is a chance of missing events, but it's not clear to me how that would be possible in this case.  I'm pretty new to the code, so maybe @smarterclayton, @ramr or @pmorie could chime in?